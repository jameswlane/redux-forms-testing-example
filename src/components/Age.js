import React from 'react'
import { Field } from "redux-form";
import { RenderField } from "./RenderField";
import { minValue, number, required, tooYoung } from "../validation";

export function Age() {
  return <Field
    name="age"
    type="number"
    component={RenderField}
    label="Age"
    validate={[required, number, minValue]}
    warn={tooYoung}
  />;
}
