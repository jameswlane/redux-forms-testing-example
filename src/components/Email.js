import React from 'react'
import { Field } from "redux-form";
import { RenderField } from "./RenderField";
import { aol, email } from "../validation";

export function Email() {
  return <Field
    name="email"
    type="email"
    component={RenderField}
    label="Email"
    validate={email}
    warn={aol}
  />;
}
