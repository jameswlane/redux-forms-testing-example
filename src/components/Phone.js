import React from 'react'
import { Field } from "redux-form";
import { RenderField } from "./RenderField";
import { phoneNumber, required } from "../validation";

export function Phone() {
  return <Field
    name="phone"
    type="number"
    component={RenderField}
    label="Phone number"
    validate={[required, phoneNumber]}
  />;
}
