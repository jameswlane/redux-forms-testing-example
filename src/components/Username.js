import React from 'react'
import { Field } from "redux-form";
import { RenderField } from "./RenderField";
import { alphaNumeric, maxLength, minLength, required } from "../validation";

export function Username() {
  return <Field
    name="username"
    type="text"
    component={RenderField}
    label="Username"
    validate={[required, maxLength, minLength]}
    warn={alphaNumeric}
  />;
}
