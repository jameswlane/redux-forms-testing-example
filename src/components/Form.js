import React from 'react'

import { Username } from './Username';
import { Email } from './Email';
import { Age } from './Age';
import { Phone } from './Phone';

export const Form = props => {
  const { handleSubmit, pristine, reset, submitting } = props;
  return (
    <form onSubmit={handleSubmit}>
      <Username/>
      <Email/>
      <Age/>
      <Phone/>
      <div>
        <button type="submit" disabled={submitting}>
          Submit
        </button>
        <button type="button" disabled={pristine || submitting} onClick={reset}>
          Clear Values
        </button>
      </div>
    </form>
  )
};
