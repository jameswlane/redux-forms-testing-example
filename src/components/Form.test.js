import React from 'react'
import { cleanup, render } from 'react-testing-library'

import { Form } from './Form';
import * as Username from './Username';
import * as  Email from './Email';
import * as Age from './Age';
import * as  Phone from './Phone';

describe('Component => RenderField', () => {
  beforeEach(() => {
    jest.spyOn(Username, 'Username').mockImplementation(() => (<span>Username</span>));
    jest.spyOn(Email, 'Email').mockImplementation(() => (<span>Email</span>));
    jest.spyOn(Age, 'Age').mockImplementation(() => (<span>Age</span>));
    jest.spyOn(Phone, 'Phone').mockImplementation(() => (<span>Phone</span>));
  })
  afterEach(() => {
    cleanup();
    jest.restoreAllMocks();
  });
  it('should match snapshot', () => {
    const props = {
      handleSubmit: jest.fn(),
      pristine: true,
      reset: jest.fn(),
      submitting: false
    };

    const { container } = render(<Form {...props}/>);

    expect(container.firstChild).toMatchSnapshot();
  })
});


