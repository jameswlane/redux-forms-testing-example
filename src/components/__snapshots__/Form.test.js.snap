// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component => RenderField should match snapshot 1`] = `
<form>
  <span>
    Username
  </span>
  <span>
    Email
  </span>
  <span>
    Age
  </span>
  <span>
    Phone
  </span>
  <div>
    <button
      type="submit"
    >
      Submit
    </button>
    <button
      disabled=""
      type="button"
    >
      Clear Values
    </button>
  </div>
</form>
`;
